lib/1.4/utility.dml

```diff
@@ -1212,3 +1216,181 @@ template bank_obj is (bank, name) {
         return _bank_obj();
     }
 }
+
+/**
+   ## Connect related templates
+
+   ### map\_target
+
+   A `connect` object can instantiate the template `map_target`. The template
+   defines a default implementation of `validate` which verifies that the
+   object can be used to create a map target, i.e. the Simics API
+   `SIM_new_map_target` returns a valid pointer. The template provides a
+   session variable `map_target` of the type `map_target_t *` which can
+   be used to issue transactions to the connected object.
+
+   The template defines the following methods:
+   <dl>
+*/
+template map_target is (connect, _qname) {
+    session map_target_t *map_target;
+
+    interface ram { param required = false; }
+    interface rom  { param required = false; }
+    interface io_memory  { param required = false; }
+    interface port_space { param required = false; }
+    interface translator { param required = false; }
+    interface transaction_translator { param required = false; }
+    interface transaction { param required = false; }
+    interface memory_space { param required = false; }
+
+    method validate(conf_object_t *obj) -> (bool) default {
+        if (port) {
+            SIM_attribute_error(
+                "port interfaces are not supported, use port objects instead");
+            return false;
+        }
+        local map_target_t *tmp = SIM_new_map_target(obj, NULL, NULL);
+        if (!tmp) {
+            SIM_clear_exception();
+            SIM_attribute_error(SIM_last_error());
+            return false;
+        }
+        SIM_free_map_target(tmp);
+        return true;
+    }
+
+    method set(conf_object_t *obj) {
+        SIM_free_map_target(this.map_target);
+        default(obj);
+        this.map_target = obj ? SIM_new_map_target(obj, NULL, NULL) : NULL;
+    }
+
+    /**
+       <dt><small>
+       read(uint64 addr, uint64 size) -&gt; (uint64) throws
+       </small></dt>
+       <dd>
+       Reads `size` bytes starting at `addr` in the connected object. Size
```

Comment:
Did you look at the resulting docs? My understanding is that you are in HTML land now, so backticks don't work. A double newline is needed to enter markdown land again (at the cost of some extra vertical space in the output, unfortunately).

Since mixing md and plain html is so error-prone, it's often better to reformulate the docs so that pure markdown constructs are sufficient; in this case you might instead express it as a dotted list of methods, or as separate subsections.

Reply:
I tried to follow the notation in `dml-builtins.dml`, wasn't aware of the need for double newline.
Documentation will be difficult to read if different sections in the same document use different formatting for the same logical concepts. Our markdown-strategy really feels awkward.

Reply:
Fixed.

![image](https://user-images.githubusercontent.com/82387192/145199993-554f079c-7b6e-4f17-897c-077f223ddfaf.png)

Reply:
Looks good! I agree it's awkward. Conradi has mentioned that it *can* be solved by turning dt into `### headings` and using CSS to make them look like definition lists, but that's outside of our control. Bullet lists is probably the best we can do today. Existing dl:s are leftovers from automatic migration.